CesiumRasterOverlays/src/RasterOverlayUtilities.cpp

```diff
@@ -1596,14 +1684,62 @@ bool upsamplePrimitiveForRasterOverlays(
     const std::string_view& textureCoordinateAttributeBaseName,
     int32_t textureCoordinateIndex,
     const CesiumGeospatial::Ellipsoid& ellipsoid) {
-  if (primitive.mode != MeshPrimitive::Mode::TRIANGLES ||
-      primitive.indices < 0 ||
-      primitive.indices >= static_cast<int>(parentModel.accessors.size())) {
-    // Not indexed triangles, so we don't know how to divide this primitive
+  if (primitive.mode != MeshPrimitive::Mode::TRIANGLES &&
+      primitive.mode != MeshPrimitive::Mode::TRIANGLE_FAN &&
+      primitive.mode != MeshPrimitive::Mode::TRIANGLE_STRIP) {
+    // Not triangles, so we don't know how to divide this primitive
     // (yet). So remove it.
     return false;
   }
 
+  if (primitive.indices < 0 ||
+      primitive.indices > static_cast<int64_t>(parentModel.accessors.size())) {
+    const auto& positionIt = primitive.attributes.find("POSITION");
+    if (positionIt == primitive.attributes.end()) {
+      // No position buffer - nothing we can do here
+      return false;
+    }
+
+    // No indices buffer - pick the smallest indices type that will fit all
+    // vertices
+    const int64_t vertexCount =
+        parentModel.accessors[static_cast<size_t>(positionIt->second)].count;
```

Comment:
I don't see anything handling the possibility of an out-of-range array access here, if the accessor referenced by the POSITION attribute is invalid. `parentModel.getSafe` may be useful.